### PR TITLE
Add section on Astronomer Registry to Docs

### PR DESCRIPTION
**Pull request by paolaperaza**
To address #192, opening up this PR to add a new section in our [Develop your Project doc](https://docs.astronomer.io/cloud/develop-project) called `Browse Airflow Providers and Modules` that:

- Briefly describes and links out to Astronomer Registry
- Has 3 bullet points with main 3 things you can reference the Registry for
- Brief line at the end connecting the registry to the rest of this doc

@jwitz @virajmparekh Do you two want to take a look at this and LMK what you think? TBH I think in the future this can go into some sort of "Write a DAG" tutorial, but I felt like this was the best place for it for now. Acknowledging that it's the least "specific" or "action oriented" section in this doc

### CODE REVIEW

**Review by virajmparekh — APPROVED** [review]

File: cloud/develop-project.md [hunk 1]
```diff
@@ -91,6 +91,18 @@ All changes made to the following files require rebuilding your image:
 
 To rebuild your project after making a change to any of these files, you must [restart your local environment](develop-project.md#restart-your-local-environment).
 
+## Explore Airflow Providers and Modules
+
+As you customize your Astronomer Project and expand your use case for Airflow, we recommend exploring the [Astronomer Registry](https://registry.astronomer.io/), a central library of Airflow modules, providers and DAGs that become the building blocks for your DAGs.
```
Comment (jwitz) [inline cloud/develop-project.md:96]:
```suggestion
As you customize your Astronomer Project and expand your use case for Airflow, we recommend exploring the [Astronomer Registry](https://registry.astronomer.io/), a library of Airflow modules, providers, and DAGs that serve as the building blocks for data pipelines.
```

File: cloud/develop-project.md [hunk 2]
```diff
@@ -91,6 +91,18 @@ All changes made to the following files require rebuilding your image:
...
+
+The Astronomer Registry includes:
+
+- Example DAGs for many data sources and destinations. For instance, an example DAG that [showcases a data quality use case with Snowflake and Great Expectations](https://registry.astronomer.io/dags/simple-great-expectations-snowflake-el).
```
Comment (jwitz) [inline cloud/develop-project.md:100]:
```suggestion
- Example DAGs for many data sources and destinations. For example, you can build out a data quality use case for Snowflake and Great Expectations based on the [Great Expectations Snowflake Example DAG](https://registry.astronomer.io/dags/simple-great-expectations-snowflake-el).
```

File: cloud/develop-project.md [hunk 3]
```diff
@@ -91,6 +91,18 @@ All changes made to the following files require rebuilding your image:
...
+- Documentation for Airflow providers, such as [Databricks](https://registry.astronomer.io/providers/databricks), [Snowflake](https://registry.astronomer.io/providers/snowflake), and [Postgres](https://registry.astronomer.io/providers/postgres). This documentation is comprehensive and based on Airflow source code.
+- Documentation for Airflow modules, such as the [PythonOperator](https://registry.astronomer.io/providers/apache-airflow/modules/pythonoperator), [BashOperator](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator), or [S3ToRedshiftOperator](https://registry.astronomer.io/providers/amazon/modules/s3toredshiftoperator). These modules include guidance on how to set Airflow Connections and parameters for them.
```
Comment (jwitz) [inline cloud/develop-project.md:102]:
```suggestion
- Documentation for Airflow modules, such as the [PythonOperator](https://registry.astronomer.io/providers/apache-airflow/modules/pythonoperator), [BashOperator](https://registry.astronomer.io/providers/apache-airflow/modules/bashoperator), and [S3ToRedshiftOperator](https://registry.astronomer.io/providers/amazon/modules/s3toredshiftoperator). These modules include guidance on how to set Airflow connections and their parameters. 
```

**Review by jwitz — APPROVED** [review]
V minor copy edits, otherwise LGTM!

**Comment by paolaperaza** [comment]
Amaze - thanks team. Merging!

**Pull request merged**